File: jimmer-ddd-java-spring-boot-starter/src/test/java/org/morecup/jimmerddd/java/spring/domain/BaseEntity.java
package org.morecup.jimmerddd.java.spring.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.babyfish.jimmer.sql.GeneratedValue;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.MappedSuperclass;

import java.time.LocalDateTime;

@MappedSuperclass
public interface BaseEntity {

    /**
     * 获取实体的唯一标识
     * @return 实体的唯一标识
     */
    @Id
    @GeneratedValue(generatorType = SnowflakeIdGenerator.class)
    long getId();

    /**
     * 获取创建时间
     * df(2025-01-20 10:00:00)
     * @return 创建时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    LocalDateTime getCreateTime();

    /**
     * 获取修改时间
     * df(2025-01-20 10:00:00)
     * @return 修改时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    LocalDateTime getUpdateTime();

    /**
     * 获取创建者id
     * @return 创建者id
     */
    Long getCreateBy();

    /**
     * 获取修改者id
     * @return 修改者id
     */
    Long getUpdateBy();
}

File: Jimmer-ddd-kotlin-spring-boot-starter/src/test/kotlin/org/morecup/jimmerddd/kotlin/spring/domain/BaseEntity.kt
package org.morecup.jimmerddd.kotlin.spring.domain

import com.fasterxml.jackson.annotation.JsonFormat
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.MappedSuperclass
import java.time.LocalDateTime

/*
 * see CommonEntityDraftInterceptor
 */
@MappedSuperclass
interface BaseEntity {

    @Id
    @GeneratedValue(generatorType = SnowflakeIdGenerator::class)
    val id: Long

    /**
     * 创建时间
     * df(2025-01-20 10:00:00)
     */
    @get:JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    val createTime: LocalDateTime

    /**
     * 修改时间
     * df(2025-01-20 10:00:00)
     */
    @get:JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    val updateTime: LocalDateTime

    /**
     * 创建者id
     */
    val createBy: Long?

    /**
     * 修改者id
     */
    val updateBy: Long?
}